### PR TITLE
lightdm-slick-greeter: init at 1.5.9

### PR DESCRIPTION
**Pull request by water-sucks**
###### Description of changes

Didn't see this packaged and I wanted to use it for my LightDM greeter. I think a lot of people are accustomed to using this because it is the default on Linux Mint and whatnot.

###### Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [X] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [X] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: nixos/modules/services/x11/display-managers/lightdm-greeters/slick.nix
Comment (IvarWithoutBones) [inline nixos/modules/services/x11/display-managers/lightdm-greeters/slick.nix:342]: Could an INI generator be used instead with a `settings` option? Manually including anything configurable seems like a lot of effort to maintain.

File: pkgs/applications/display-managers/lightdm-slick-greeter/default.nix
Comment (IvarWithoutBones) [inline pkgs/applications/display-managers/lightdm-slick-greeter/default.nix:26]: This shouldn't refer to a random git revision, but instead a release tag. If that does not exist, we follow the following format for `version`: `0.pre+<date>` where `<date>` is the date of the commit.

File: pkgs/applications/display-managers/lightdm-slick-greeter/default.nix
Comment (IvarWithoutBones) [inline pkgs/applications/display-managers/lightdm-slick-greeter/default.nix:34]: Could you add a comment explaining why this is done?

File: pkgs/top-level/all-packages.nix
Comment (IvarWithoutBones) [inline pkgs/top-level/all-packages.nix:30388]: This isn't required, all packages from `xorg` should be in scope already.
```suggestion
```

**Review by IvarWithoutBones — CHANGES_REQUESTED** [review]

File: pkgs/top-level/all-packages.nix
Comment (bobby285271) [inline pkgs/top-level/all-packages.nix:30387]: Not sure if `lightdm-slick-greeter` will be a better attribute name? (FYI I am renaming lightdm_gtk_greeter to lightdm-gtk-greeter in #184454)

Comment (water-sucks) [reply pkgs/top-level/all-packages.nix:30387]: Yeah, I was a bit confused on whether or not I should use underscores or dashes. I'll change it to that.

File: pkgs/applications/display-managers/lightdm-slick-greeter/default.nix
Comment (water-sucks) [reply pkgs/applications/display-managers/lightdm-slick-greeter/default.nix:34]: Got it; the configure was failing during the autoreconfPhase.

File: nixos/modules/services/x11/display-managers/lightdm-greeters/slick.nix
Comment (water-sucks) [reply nixos/modules/services/x11/display-managers/lightdm-greeters/slick.nix:342]: I think this is probably better, yeah, but I'm not quite sure which convention to follow, because I don't think the INI generator converts camel case to dashes?  I'll do some research and try to do this though.

**Comment by water-sucks** [comment]
Did the requested changes. I had to make a minor change to add actual support for numlockx and remove the hardcoded path I didn't notice before.

File: maintainers/maintainer-list.nix
Comment (bobby285271) [inline maintainers/maintainer-list.nix:14845]: Sorry to be nitpicky, though there are some entries ignoring the request, it will be nice to keep the list alphabetically sorted.

https://github.com/NixOS/nixpkgs/blob/f0ec9e1f968bbeb214dd0c92989dc8a8beb7a724/maintainers/maintainer-list.nix#L49

File: pkgs/applications/display-managers/lightdm-slick-greeter/default.nix
Comment (bobby285271) [inline pkgs/applications/display-managers/lightdm-slick-greeter/default.nix:18]: Why is this off by default? (TBH I will probably just drop this option and set that in `slick-greeter.conf` if numlockx does not have huge impact in closure so we don't have to build both when testing)

File: pkgs/applications/display-managers/lightdm-slick-greeter/default.nix
Comment (bobby285271) [inline pkgs/applications/display-managers/lightdm-slick-greeter/default.nix:28]:
```suggestion
    rev = version;
```

File: pkgs/applications/display-managers/lightdm-slick-greeter/default.nix
Comment (bobby285271) [inline pkgs/applications/display-managers/lightdm-slick-greeter/default.nix:40]: We can probably move this to `preAutoreconf`.

File: pkgs/applications/display-managers/lightdm-slick-greeter/default.nix
Comment (bobby285271) [inline pkgs/applications/display-managers/lightdm-slick-greeter/default.nix:44]: And probably move this to postPatch. We can do this unconditionally since we can disable this in slick-greeter.conf

**Review by bobby285271 — COMMENTED** [review]
(Sorry I don't really review modules so it will be nice to get some review for that before merging)

File: pkgs/applications/display-managers/lightdm-slick-greeter/default.nix
Comment (bobby285271) [inline pkgs/applications/display-managers/lightdm-slick-greeter/default.nix:35]: Maybe we don't need numlockx in buildInputs at all if it is only called once in `src/slick-greeter.vala`

File: pkgs/applications/display-managers/lightdm-slick-greeter/default.nix
Comment (bobby285271) [inline pkgs/applications/display-managers/lightdm-slick-greeter/default.nix:50]: A quick search of `indicator-services-command` does not return any result for me, why is this needed? (Maybe also be nice to check other flags as well)

File: pkgs/applications/display-managers/lightdm-slick-greeter/default.nix
Comment (water-sucks) [reply pkgs/applications/display-managers/lightdm-slick-greeter/default.nix:40]: Wow, I didn't know this existed. I couldn't find anything about `autoreconfPhase` except for a small section in the nixpkgs manual.

File: pkgs/applications/display-managers/lightdm-slick-greeter/default.nix
Comment (water-sucks) [reply pkgs/applications/display-managers/lightdm-slick-greeter/default.nix:50]: Oops, forgot to take this out. I was mostly modifying the derivation I saw from lightdm-gtk-greeter; seems to have had no effect though.

**Comment by bobby285271** [comment]
I took the liberty to force push some changes, tested this in a VM and seems to work. I will do some more tests in a few days.

**Comment by water-sucks** [comment]
@bobby285271 thanks for cleaning up the PR! I definitely did miss a few buildInputs for the scripts and other things in the derivation (I only checked if the package was running successfully), and didn't know if I had to include all the options for it in the module or not; including all of the options instead of just leaving it does seem like a nightmare to maintain, now that I think about it.

Also pretty cool that Cinnamon switched to slick-greeter as the default; changing default behavior isn't something I wanted to do without approval or something, though, so I held off on that. Nice to see it here though 😄

**Comment by bobby285271** [comment]
[Mint defaults to this greeter since 18.2](https://blog.linuxmint.com/?p=3235) so I though we can probably follow what upstream does. I don't have preferences on greeters and/or DMs (I am actually using Pantheon greeter in most cases since I maintain Pantheon) so I won't mind to revert those changes if requested.

**Comment by mkg20001** [comment]
Changing to whatever mint uses is fine. I just newer switched to that greeter since it wasn't ported and I didn't have the resources for that.

**Review by IvarWithoutBones — APPROVED** [review]

**Comment by water-sucks** [comment]
Is there anything else that needs to be done? Or are we okay to merge?

**Comment by IvarWithoutBones** [comment]
> Is there anything else that needs to be done? Or are we okay to merge?

As far as Im concerned this is good to merge, just waiting on others who reviewed to take another look.

**Review by bobby285271 — APPROVED** [review]
Tested with multi users and multi sessions. ~~I don't have new stuff to address so far.~~

---

Edited on 20, Aug: Channel blocker fix at https://github.com/NixOS/nixpkgs/pull/187564

**Pull request merged**
